lib/deimos/instrumentation.rb

```diff
@@ -80,4 +80,9 @@ def self.send_produce_error(event)
     event = ActiveSupport::Notifications::Event.new(*args)
     KafkaListener.send_produce_error(event)
   end
+
+  ActiveSupport::Notifications.subscribe('batch_consumption.invalid_records') do |*args|
+    payload = ActiveSupport::Notifications::Event.new(*args).payload
+    payload[:consumer].process_invalid_records(payload[:records])
```

Comment:
Calling code should be able to do this themselves rather than us having a method that we call explicitly.